lib-es5/api_client/execute_request.js

```diff
@@ -79,16 +87,17 @@ function execute_request(method, params, auth, api_url, callback) {
           result.error.http_code = res.statusCode;
         }
 
-        if (result.error) {
+        if (result.error && !callback) {
           deferred.reject(Object.assign({
             request_options,
             query_params
           }, result));
-        } else {
-          deferred.resolve(result);
         }
-        if (typeof callback === "function") {
+
+        if (callback) {
           callback(result);
+        } else {
+          deferred.resolve(result);
```

Comment:
Why not calling `handleRequestError`?

Reply:
That's resolving.. it's not the same behavior